packages/omgidl-serialization/src/MessageReader.test.ts

```diff
@@ -1319,4 +1319,18 @@ module builtin_interfaces {
 
     expect(msgout).toEqual(data);
   });
+  it("Reads struct with no members", () => {
+    const msgDef = `
+      struct Message {
+      };
+    `;
+    const data = {};
+    const writer = new CdrWriter({ kind: EncapsulationKind.PL_CDR_LE });
+    writer.sentinelHeader(); // end of struct
```

Comment:
Not sure if there is more to write than just the header? @snosenzo

Reply:
I think it because it doesn't have `@mutable` it actually doesn't even have a sentinelHeader. SHould probably also make one with mutable and it would have an `emHeader` and a `sentinelHeader`

Reply:
> SHould probably also make one with mutable and it would have an emHeader and a sentinelHeader

I added one with `@mutable`, but it fails if I write an `emHeader` :thinking:

Reply:
I feel like it should be able to take an emHeader 🤔

Reply:
oh wait nevermind, `emheaders` are for _members_ not the structs themselves. The root structs can have `dHeaders` however, but that's in PL_CDR2 iirc.